client/src/services/userService.test.js

```diff
@@ -0,0 +1,38 @@
+import { resourceStub } from "../../setupTests";
+
+import UserService from "./userService";
+
+describe("UserService", () => {
```

Comment:
Fun fact: if you comment out or delete the whole implementation of the actual `getByUser` method, these tests still pass. You're replacing the real implementation with a test double, the mock function created with `jest.fn()`, then calling _the test double_ and asserting on its behaviour (i.e. _"does the test double do what I literally just asked it to do?"_) This is why if you look at the test coverage report a big chunk of the service simply isn't covered by the tests.

Reply:
FYI this continues to have the problem that you're testing a test double; you've replaced the entire real implementation with `jest.mock("./userService");`. It's absolutely crucial when writing tests to ensure that your actual implementation is involved - even if you write the tests afterwards for code that's (or that you think is) working, temporarily break the implementation _on purpose_ to watch them fail. Test doubles are for _collaborators_, things the system under test uses, not to entirely replace the thing you're supposed to be testing.